index.html

```diff
@@ -559,53 +444,59 @@ <h3>
+          <a data-xref-for="list">is empty</a>, reject |promise| with a
+          "{{NotAllowedError}}" {{DOMException}} and return |promise|.
           </li>
-          <li>If |lock|.{{[[requestPromise]]}} is not `undefined`,
-            reject |promise| with a "{{InvalidStateError}}" {{DOMException}},
-            and run <a>abort the request steps</a>.
+          <li>If the <a>current global object</a> is the {{Window}} object and
+          the |document|'s [=Document/browsing context=] is `null`, reject
+          |promise| with a "{{NotAllowedError}}" {{DOMException}} and
+          return |promise|.
           </li>
-          <li>If |options|.|signal:AbortSignal|’s <a>aborted flag</a> is set, then
-          reject |promise| with an "{{AbortError}}" {{DOMException}},
-          and run <a>abort the request steps</a>.
+          <li data-tests="wakelock-disabled-by-feature-policy.https.sub.html">
+          If |document| is not <a>allowed to use</a> the <a>policy-controlled
+          feature</a> named "`wake-lock`", reject |promise| with a
+          "{{NotAllowedError}}" {{DOMException}} and return |promise|.
+          </li>
+          <li>If the <a>user agent</a> <a data-lt="deny wake lock">denies the
+          wake lock</a> of this |type| for |document|, reject |promise|
+          with a "{{NotAllowedError}}" {{DOMException}} and return |promise|.
           </li>
-          <li>If |options|.|signal| is not `null`, then [=AbortSignal/add=] to |options|.|signal|:
+          <li>If |options|.|signal:AbortSignal| member is present, then:
             <ol>
-              <li>Run <a>release a wake lock</a> on |lock|.
+              <li>If |options|.|signal|’s <a>aborted flag</a> is set, then
+                reject |promise| with an "{{AbortError}}" {{DOMException}},
+                and return |promise|.
+              </li>
+              <li>
+                [=AbortSignal/Add=] to |options|.|signal|:
+                <ol>
+                  <li>Run <a>release a wake lock</a> with |promise| and |type|.
+                  </li>
+                </ol>
               </li>
             </ol>
           </li>
-          <li>Set |lock|.{{[[requestPromise]]}} to promise.
-          </li>
           <li>Return |promise| and run the following steps <a>in parallel</a>:
             <ol>
               <li>Let |state:PermissionState| be the result of running and waiting for the <a>
-              obtain permission</a> steps with |lock|.|type:WakeLockType|.
+              obtain permission</a> steps with |type|.
               </li>
               <li>If |state| is {{"denied"}}, then reject |promise| with a
-              "{{NotAllowedError}}" {{DOMException}}, and run <a>abort the
-              request steps</a>.
+              "{{NotAllowedError}}" {{DOMException}}, and abort these steps.
               </li>
               <li>If the <a>current global object</a> is not a
```

Comment:
Can't you move all these "not a DedicatedWorkerGlobalScope" checks up and merge them with step 4 ("If the current global object is the Window object and the document's browsing context is null [...]")?

Reply:
I don't see how that becomes simpler, but maybe you can show me what you are thinking?

Reply:
Step 4 covers "the current global object is not a DedicatedWorkerGlobalScope, so check document's browsing context is not null". To me, it sounds simpler to expand that to "also, if it's not null, check it is fully active _and_, if `type` is 'screen', make sure the page is not hidden" (unless that really needs to be done arbitrarily later after `promise` has already been returned), otherwise we have step 4 and then another "if the current global object is not a DedicatedWorkerGlobalScope, do this and that" section that in any case would also probably need to check the browsing context is not null again.

Reply:
done